### ISSUE TITLE
win recorders, reduce the number of time system calls

### ISSUE THREAD

**Issue by sparvu** [kronometrix/recording#86]
In order to make things even easier we could reduce the way we calculate elapsed times inside data recorders. Currently there is an extra call on gettimeofday() without no reason at all.

**Comment by sparvu** [comment]
proposed method:

```
    # end payload
    my $tnow = [gettimeofday];
    my $elapsed = tv_interval ($start, $tnow);

    # we default to interval
    my $stime = 1;
    if ( $elapsed < $interval ) { $stime = $interval - $elapsed; }
    else {
        if (RunningAsService()) {
            writelog ("warn: elapsed time > interval, elapsed=$elapsed(s), interval=$interval(s)");
	} else {
	    if ($interval > 1) {
            writelog ("warn: elapsed time > interval, elapsed=$elapsed(s), interval=$interval(s)") if ($logging);
            print "warn: elapsed time > interval, elapsed=$elapsed(s), interval=$interval(s)\n"; }
	}
    }

```

**Comment by sparvu** [comment]
As well in the future I need to check to make more efficient and take less time getting the system metrics. Currently this takes a bit of time.

**Comment by sparvu** [comment]
fixed in 1.0.19 RC1
